### DISCUSSION TITLE
Transforms state from symmetry-reduced basis to full (symmetry-free) basis.

### DISCUSSION THREAD

**Discussion by tpyang** [QuSpin/QuSpin#571] [Q&A] [answered]
Since the operator can not be carried out in the symmetry-reduced basis, I have to transfer the state from symmetry-reduced basis to a full (symmetry-free) basis. 

But the full basis is chosen to contain all possible states (all particle numbers and spin configuration), and is extremely large. Is it possible to transfer the symmetry-reduced basis to a full basis with a certain quantum number (as follows, given Nf)?
basis_2d=spinful_fermion_basis_general(N_2d,Nf=(Nup,Ndn), double_occupancy=False)

**Comment by tpyang** [comment]
Or can we simply add all symmetry basis sectors to make a full basis? Will be great if there is such a function.

**Comment by mgbukov** [comment]
check out the [`basis.project_from`](https://weinbe58.github.io/QuSpin/generated/quspin.basis.spinful_fermion_basis_general.html#quspin.basis.spinful_fermion_basis_general.project_from) function, with its `pcon` option. This allows you to keep the partice sector fixed. 

It's not possible to keep an arbitrary symmetry fixed, but you can then apply `basis.project_to` of another basis object to reduce to the corresponding sector.

**Reply by tpyang** [reply]
Thank you for the quick reply! I tried this option with [basis.project_from](https://weinbe58.github.io/QuSpin/generated/quspin.basis.spinful_fermion_basis_general.html#quspin.basis.spinful_fermion_basis_general.project_from), but the output full basis assumes double occupancy by default. Therefore, basis.project_to doesn't work, either. 

Can I construct a new basis from the combination of all symmetry sector basis? Any example for manually constructing a spinful_fermion_basis_general basis from a list of integers?

**Comment by mgbukov** [comment, marked as answer]
yeah, check out [this example](https://weinbe58.github.io/QuSpin/examples/example16.html#example16-label) and the [user_basis tutorial](https://weinbe58.github.io/QuSpin/user_basis.html).

**Reply by tpyang** [reply]
Thank you! I solved my problem by adding double_occupancy=self._double_occupancy to the project_from function.

**Comment by mgbukov** [comment]
From your response, it looks like the double_occupancy flag was missing in project_from, is this correct?. Did we forget to take it into account? If yes, can I ask you to add it to fix this bug also in the project_to function and make a pull request into the most recent dev branch, so other users don't run into the same issue? Thanks!

**Reply by tpyang** [reply]
Yes, this flag is missing in project_from and project_to. 
Fro example line 526 and 628 in the most recent dev branch.
https://github.com/weinbe58/QuSpin/blob/dev_0.3.7/quspin/basis/basis_general/base_general.py

I will make a pull request for this bug.
